conans/client/build/cmake_flags.py

```diff
@@ -375,16 +375,16 @@ def get_definitions(self, cmake_version):
 
         # Adjust automatically the module path in case the conanfile is using the
         # cmake_find_package or cmake_find_package_multi
-        install_folder = self._conanfile.install_folder.replace("\\", "/")
+        generators_folder = self._conanfile.generators_folder.replace("\\", "/")
```

Comment:
```suggestion
        generators_folder = self._conanfile.layout.generators_folder.replace("\\", "/")
```

Confused about when to use it from ``layout`` or from ``conanfile`` directly.

Reply:
Always with `conanfile` is the rule. Only when declaring layout except for: Special cases (have to be checked one by one), (some bugs LoL), and internal logic should access to the `conanfile.layout.xxx_folder`.

Reply:
And that's because the `conanfile.xxx_folder` prepends the correct root folder, the other one is just relative, so usually is not something to be used.